crates/taplo-common/src/schema/mod.rs

```diff
@@ -543,9 +543,12 @@ impl<E: Environment> Schemas<E> {
         if let Some(all_ofs) = schema["allOf"].as_array() {
             if !all_ofs.is_empty() && composed {
                 let mut schema = schema.clone();
-                if let Some(obj) = schema["allOf"].as_object_mut() {
-                    obj.remove("allOf");
-                }
+                // Remove the allOf because we're resolving in this spot -- not doing so will cause
+                // infinite recursion.
+                schema
+                    .as_object_mut()
+                    .expect("schema is an object")
+                    .remove("allOf");
 
                 let mut merged_all_of = Value::Object(serde_json::Map::default());
 
```

Comment:
I noticed below in line 569, and above in line 512/518/525, that `depth` wasn't getting subtracted by 1. Is this expected? Naively I'd expect `depth` to be subtracted here.

Reply:
Ah, it looks like https://github.com/tamasfe/taplo/issues/353 is related.